### DISCUSSION TITLE
Stream request body with "request.request.pipe(...)"

### DISCUSSION THREAD

**Discussion by wiliam-paradox** [adonisjs/core#3381] [General]
### Descripcion

It's possible to disable body parser for a specific route or for all the project and stream the request body in the controller?

### Example

```typescript
async controllerMethod({request, response} : HttpContextContract) {
    let writeStream = fs.createWriteStream('request-data.json');

    let req = request.request;
    req.pipe(writeStream)
    req.on('end', () => {
         response.json(true);
    });
}
```

### Objective

We are using AdonisJS 5 as a Reserve proxy with some logic, we want to accept big requests but not to load them into memory, so we have disabled JSON parser, but request.raw() holds all the string request in memory.

We want to stream the request to Google Storage and to the Backend Server and we need to do it after entering the controller, where the logic to what to do with the request is done.

In another small frameworks like Express it's possible:

https://stackoverflow.com/questions/61660783/nodejs-expressjs-how-to-stream-request-body-without-buffering

I imagine that the body parser module reads the request content and pipe can't be consumed anymore.

Thanks for your time.

**Comment by wiliam-paradox** [comment]
Ok, after looking at this file (bodyparser module):

https://github.com/adonisjs/bodyparser/blob/develop/src/BodyParser/index.ts

I have seen that I can disable it at all emptying whitelistedMethods:

![2021-11-29_17-25_7865](https://user-images.githubusercontent.com/29425623/143905252-a07b56df-f4fc-4523-a39b-76dbdb26f0c6.png)

At the `config/bodyparser.ts`:

```ts
const bodyParserConfig: BodyParserConfig = {
    /*
    |--------------------------------------------------------------------------
    | White listed methods
    |--------------------------------------------------------------------------
    |
    | HTTP methods for which body parsing must be performed. It is a good practice
    | to avoid body parsing for `GET` requests.
    |
    */
    // whitelistedMethods: ['POST', 'PUT', 'PATCH', 'DELETE'],
    whitelistedMethods: [],
```

You can disable processing request at multipart, raw, json... individually. Per route only available at multipart, it will be nice to have it not only in multipart but in json and raw too.

**Comment by wiliam-paradox** [comment]
Or disable the middleware at all at `start/kernel.ts`

```ts
Server.middleware.register([
  // () => import('@ioc:Adonis/Core/BodyParser'),
])
```

**Comment by thetutlage** [comment]
Yes disable the bodyparser middleware and self handle the request stream.
